plugins/unix-socket.js

```diff
@@ -10,13 +10,19 @@ var started = false
 module.exports = function (opts) {
   const socket = path.join(opts.path || '', 'socket')
   const addr = 'unix:' + socket
-  const scope = opts.scope || 'device'
+  let scope = opts.scope || 'device'
   opts = opts || {}
   return {
     name: 'unix',
     scope: function() { return scope },
-    server: !opts.server ? null : function (onConnection, cb) {
-      if(started) return
+    server: function (onConnection, cb) {
+      if (started) return
+
+      if (scope !== "device") {
+        debug('Insecure scope for unix socket! Reverting to device scope')
```

Comment:
My personal preference here would be to either throw an error or print some logging output to note the config problem. I don't often run with `DEBUG=*` or `DEBUG=multiserver:*` so I'd never see this message. If you feel strongly about keeping it as `debug()` I'm happy to trust your decision here. :+1:

Reply:
The problem as dominic pointed out was that if we let this continue, then we have a security problem. So either we totally bail here, or we force the device option. I think the last option is the better. Also the use of debug is to stay consistant with the rest of the package, see this [commit](https://github.com/ssbc/multiserver/commit/785de927772e1cdf22b7b8331f2f4d7686afc9aa).

Reply:
Right, I'm on the same page. My preference is that either of these are *loud*, although I don't really mind which one we choose. My concern is that if we silently change configs then someone may learn "`local` doesn't get broadcast over the network" and set up an insecure network-accessible interface.

I like `debug`, but I think this is something that needs to be logged rather than *only* used when the user sets up `DEBUG` a certain way. If you feel strongly about using the module then maybe we could do a `debug.enabled` dance?

```js
const isEnabled = debug.enabled
debug.enabled = true
debug('Insecure scope for unix socket! Reverting to device scope')
debug.enabled = isEnabled
```

Anyway, this PR is a vast improvement over the current situation, so we can merge this regardless and discuss logging output later. Thanks for taking care of this bug. :heart:

Reply:
Thanks bundy. I agree completely with the debug option btw